src/totp/converters.ts

```diff
@@ -0,0 +1,16 @@
+export const hexToUint8Array = (hex: string): Uint8Array => {
+  const length = hex.length / 2;
+  const uintArray = new Uint8Array(length);
+  for (let i = 0; i < length; i++) {
+    uintArray[i] = parseInt(hex.substring(2 * i, 2 * i + 2), 16);
+  }
+  return uintArray;
+};
+
+export const hex2dec = (hex: string): number => {
+  return parseInt(hex, 16);
+};
+
+export const dec2hex = (dec: number): string => {
+  return (dec < 15.5 ? '0' : '') + Math.round(dec).toString(16);

```

Comment:
Same thing here - why 15.5 for example?

Reply:
I'll implement it differently to make it more readable.

Reply:
ℹ️ My previous implementation was quite smart 🙈  but unreadable at the same time. 
I've made an equivalent implementation which should work the same, and be quite self-explanatory.